pandas/core/internals/construction.py

```diff
@@ -160,7 +160,29 @@ def init_ndarray(values, index, columns, dtype=None, copy=False):
     # on the entire block; this is to convert if we have datetimelike's
     # embedded in an object type
     if dtype is None and is_object_dtype(values):
-        values = maybe_infer_to_datetimelike(values)
+
+        if values.ndim == 2 and values.shape[0] != 1:
```

Comment:
this is much more messy, can we change something else to make this nicer?

Reply:
Not really.  I'm looking into the other places where maybe_infer_to_datetimelike is used to see if some of this can go into that.  We could separate this whole block into a dedicated function.  But one way or another we need to bite the bullet.

Reply:
so the inside of list loop should be in pandas.core.dtypes.cast, no? (obviously up until you make the blocks themselves)

Reply:
I'd like to leave this for the next pass when I'm taking a more systematic look at maybe_infer_to_datetimelike